packages/commonwealth/server/util/cosmosCache.ts

```diff
@@ -0,0 +1,132 @@
+import type { Request, RequestHandler } from 'express';
+import { cacheDecorator } from '../../../common-common/src/cacheDecorator';
+
+export const cosmosCacheRPC = (duration: number): RequestHandler => {
+  return async function cacheIfCacheWorthyRPC(req, res, next) {
```

Comment:
`cacheIfCacheWorthyRPC`
-> @mhagel we can simplify this if you like
-> just add function to compute key here:
   -> [cacheKeyUtils](https://github.com/hicommonwealth/commonwealth/blob/2622373e07850abc3aa81d5f56c1473a49a72de0/packages/common-common/src/cacheKeyUtils.ts)
   
  if you return NULL from key, we wont cache it!
  Just same logic instead make it part of key generator function
  Let me know if any clarification needed!

Reply:
else keep going, we can always refactor a little bit, thanks!

Reply:
I am using this as a filter before sending requests to the cache. If there is something we don't want to cache, we just call next() to skip the cacheMiddleware.